### ISSUE TITLE
make fails - rtlsdr_set_bias_tee undefined

### ISSUE THREAD

**Issue by uphillbattle** [merbanan/rtl_433#1825]
Rapberry Pi 3B+, Raspian stretch

I'm trying to update to latest rtl_433 by building the latest version. It fails on make, complains about `rtlsdr_set_bias_tee` undefined in `sdr.c`.

I have ` librtlsdr-dev rtl-sdr librtlsdr0` installed and all are version `0.5.3-11+rpt1`.

make log below.

I have built rtl_433 several times before (latest around year ago or so) and I have never had problems.

Any ideas what's wrong?

```
~/rtl_433/build $ make
Scanning dependencies of target r_433
[  0%] Building C object src/CMakeFiles/r_433.dir/abuf.c.o
[  1%] Building C object src/CMakeFiles/r_433.dir/am_analyze.c.o
[  1%] Building C object src/CMakeFiles/r_433.dir/baseband.c.o
[  1%] Building C object src/CMakeFiles/r_433.dir/bitbuffer.c.o
[  2%] Building C object src/CMakeFiles/r_433.dir/compat_alarm.c.o
[  2%] Building C object src/CMakeFiles/r_433.dir/compat_paths.c.o
[  3%] Building C object src/CMakeFiles/r_433.dir/compat_time.c.o
[  3%] Building C object src/CMakeFiles/r_433.dir/confparse.c.o
[  4%] Building C object src/CMakeFiles/r_433.dir/data.c.o
[  4%] Building C object src/CMakeFiles/r_433.dir/data_tag.c.o
[  5%] Building C object src/CMakeFiles/r_433.dir/decoder_util.c.o
[  5%] Building C object src/CMakeFiles/r_433.dir/fileformat.c.o
[  6%] Building C object src/CMakeFiles/r_433.dir/http_server.c.o
[  6%] Building C object src/CMakeFiles/r_433.dir/jsmn.c.o
[  6%] Building C object src/CMakeFiles/r_433.dir/list.c.o
[  7%] Building C object src/CMakeFiles/r_433.dir/mongoose.c.o
[  7%] Building C object src/CMakeFiles/r_433.dir/optparse.c.o
[  8%] Building C object src/CMakeFiles/r_433.dir/output_influx.c.o
[  8%] Building C object src/CMakeFiles/r_433.dir/output_mqtt.c.o
[  9%] Building C object src/CMakeFiles/r_433.dir/pulse_analyzer.c.o
[  9%] Building C object src/CMakeFiles/r_433.dir/pulse_demod.c.o
[ 10%] Building C object src/CMakeFiles/r_433.dir/pulse_detect.c.o
[ 10%] Building C object src/CMakeFiles/r_433.dir/pulse_detect_fsk.c.o
[ 11%] Building C object src/CMakeFiles/r_433.dir/r_api.c.o
[ 11%] Building C object src/CMakeFiles/r_433.dir/r_util.c.o
[ 11%] Building C object src/CMakeFiles/r_433.dir/rfraw.c.o
[ 12%] Building C object src/CMakeFiles/r_433.dir/samp_grab.c.o
[ 12%] Building C object src/CMakeFiles/r_433.dir/sdr.c.o
/home/pi/rtl_433/src/sdr.c: In function ‘sdr_apply_settings’:
/home/pi/rtl_433/src/sdr.c:1486:21: warning: implicit declaration of function ‘rtlsdr_set_bias_tee’ [-Wimplicit-function-declaration]
                 r = rtlsdr_set_bias_tee(dev->rtlsdr_dev, biastee);
                     ^~~~~~~~~~~~~~~~~~~
[ 13%] Building C object src/CMakeFiles/r_433.dir/term_ctl.c.o
[ 13%] Building C object src/CMakeFiles/r_433.dir/util.c.o
[ 14%] Building C object src/CMakeFiles/r_433.dir/write_sigrok.c.o
[ 14%] Building C object src/CMakeFiles/r_433.dir/devices/abmt.c.o
[ 15%] Building C object src/CMakeFiles/r_433.dir/devices/acurite.c.o
[ 15%] Building C object src/CMakeFiles/r_433.dir/devices/akhan_100F14.c.o
[ 16%] Building C object src/CMakeFiles/r_433.dir/devices/alecto.c.o
[ 16%] Building C object src/CMakeFiles/r_433.dir/devices/ambient_weather.c.o
[ 16%] Building C object src/CMakeFiles/r_433.dir/devices/ambientweather_tx8300.c.o
[ 17%] Building C object src/CMakeFiles/r_433.dir/devices/ambientweather_wh31e.c.o
[ 17%] Building C object src/CMakeFiles/r_433.dir/devices/archos_tbh.c.o
[ 18%] Building C object src/CMakeFiles/r_433.dir/devices/auriol_aft77b2.c.o
[ 18%] Building C object src/CMakeFiles/r_433.dir/devices/auriol_afw2a1.c.o
[ 19%] Building C object src/CMakeFiles/r_433.dir/devices/auriol_ahfl.c.o
[ 19%] Building C object src/CMakeFiles/r_433.dir/devices/auriol_hg02832.c.o
[ 20%] Building C object src/CMakeFiles/r_433.dir/devices/blueline.c.o
[ 20%] Building C object src/CMakeFiles/r_433.dir/devices/blyss.c.o
[ 21%] Building C object src/CMakeFiles/r_433.dir/devices/brennenstuhl_rcs_2044.c.o
[ 21%] Building C object src/CMakeFiles/r_433.dir/devices/bresser_3ch.c.o
[ 21%] Building C object src/CMakeFiles/r_433.dir/devices/bresser_5in1.c.o
[ 22%] Building C object src/CMakeFiles/r_433.dir/devices/bresser_6in1.c.o
[ 22%] Building C object src/CMakeFiles/r_433.dir/devices/bresser_7in1.c.o
[ 23%] Building C object src/CMakeFiles/r_433.dir/devices/bt_rain.c.o
[ 23%] Building C object src/CMakeFiles/r_433.dir/devices/burnhardbbq.c.o
[ 24%] Building C object src/CMakeFiles/r_433.dir/devices/calibeur.c.o
[ 24%] Building C object src/CMakeFiles/r_433.dir/devices/cardin.c.o
[ 25%] Building C object src/CMakeFiles/r_433.dir/devices/cavius.c.o
[ 25%] Building C object src/CMakeFiles/r_433.dir/devices/chuango.c.o
[ 26%] Building C object src/CMakeFiles/r_433.dir/devices/cmr113.c.o
[ 26%] Building C object src/CMakeFiles/r_433.dir/devices/companion_wtr001.c.o
[ 26%] Building C object src/CMakeFiles/r_433.dir/devices/cotech_36_7959.c.o
[ 27%] Building C object src/CMakeFiles/r_433.dir/devices/current_cost.c.o
[ 27%] Building C object src/CMakeFiles/r_433.dir/devices/danfoss.c.o
[ 28%] Building C object src/CMakeFiles/r_433.dir/devices/digitech_xc0324.c.o
[ 28%] Building C object src/CMakeFiles/r_433.dir/devices/directv.c.o
[ 29%] Building C object src/CMakeFiles/r_433.dir/devices/dish_remote_6_3.c.o
[ 29%] Building C object src/CMakeFiles/r_433.dir/devices/dsc.c.o
[ 30%] Building C object src/CMakeFiles/r_433.dir/devices/ecodhome.c.o
[ 30%] Building C object src/CMakeFiles/r_433.dir/devices/ecowitt.c.o
[ 31%] Building C object src/CMakeFiles/r_433.dir/devices/efergy_e2_classic.c.o
[ 31%] Building C object src/CMakeFiles/r_433.dir/devices/efergy_optical.c.o
[ 31%] Building C object src/CMakeFiles/r_433.dir/devices/efth800.c.o
[ 32%] Building C object src/CMakeFiles/r_433.dir/devices/elro_db286a.c.o
[ 32%] Building C object src/CMakeFiles/r_433.dir/devices/elv.c.o
[ 33%] Building C object src/CMakeFiles/r_433.dir/devices/emontx.c.o
[ 33%] Building C object src/CMakeFiles/r_433.dir/devices/ert_idm.c.o
[ 34%] Building C object src/CMakeFiles/r_433.dir/devices/ert_scm.c.o
[ 34%] Building C object src/CMakeFiles/r_433.dir/devices/esa.c.o
[ 35%] Building C object src/CMakeFiles/r_433.dir/devices/esic_emt7110.c.o
[ 35%] Building C object src/CMakeFiles/r_433.dir/devices/esperanza_ews.c.o
[ 36%] Building C object src/CMakeFiles/r_433.dir/devices/eurochron.c.o
[ 36%] Building C object src/CMakeFiles/r_433.dir/devices/fineoffset.c.o
[ 36%] Building C object src/CMakeFiles/r_433.dir/devices/fineoffset_wh1050.c.o
[ 37%] Building C object src/CMakeFiles/r_433.dir/devices/fineoffset_wh1080.c.o
[ 37%] Building C object src/CMakeFiles/r_433.dir/devices/fineoffset_wh31l.c.o
[ 38%] Building C object src/CMakeFiles/r_433.dir/devices/flex.c.o
[ 38%] Building C object src/CMakeFiles/r_433.dir/devices/fordremote.c.o
[ 39%] Building C object src/CMakeFiles/r_433.dir/devices/fs20.c.o
[ 39%] Building C object src/CMakeFiles/r_433.dir/devices/ft004b.c.o
[ 40%] Building C object src/CMakeFiles/r_433.dir/devices/ge_coloreffects.c.o
[ 40%] Building C object src/CMakeFiles/r_433.dir/devices/generic_motion.c.o
[ 41%] Building C object src/CMakeFiles/r_433.dir/devices/generic_remote.c.o
[ 41%] Building C object src/CMakeFiles/r_433.dir/devices/generic_temperature_sensor.c.o
[ 41%] Building C object src/CMakeFiles/r_433.dir/devices/govee.c.o
[ 42%] Building C object src/CMakeFiles/r_433.dir/devices/gt_tmbbq05.c.o
[ 42%] Building C object src/CMakeFiles/r_433.dir/devices/gt_wt_02.c.o
[ 43%] Building C object src/CMakeFiles/r_433.dir/devices/gt_wt_03.c.o
[ 43%] Building C object src/CMakeFiles/r_433.dir/devices/hcs200.c.o
[ 44%] Building C object src/CMakeFiles/r_433.dir/devices/hideki.c.o
[ 44%] Building C object src/CMakeFiles/r_433.dir/devices/holman_ws5029.c.o
[ 45%] Building C object src/CMakeFiles/r_433.dir/devices/hondaremote.c.o
[ 45%] Building C object src/CMakeFiles/r_433.dir/devices/honeywell.c.o
[ 46%] Building C object src/CMakeFiles/r_433.dir/devices/honeywell_cm921.c.o
[ 46%] Building C object src/CMakeFiles/r_433.dir/devices/honeywell_wdb.c.o
[ 46%] Building C object src/CMakeFiles/r_433.dir/devices/ht680.c.o
[ 47%] Building C object src/CMakeFiles/r_433.dir/devices/ibis_beacon.c.o
[ 47%] Building C object src/CMakeFiles/r_433.dir/devices/ikea_sparsnas.c.o
[ 48%] Building C object src/CMakeFiles/r_433.dir/devices/infactory.c.o
[ 48%] Building C object src/CMakeFiles/r_433.dir/devices/inkbird_ith20r.c.o
[ 49%] Building C object src/CMakeFiles/r_433.dir/devices/inovalley-kw9015b.c.o
[ 49%] Building C object src/CMakeFiles/r_433.dir/devices/insteon.c.o
[ 50%] Building C object src/CMakeFiles/r_433.dir/devices/interlogix.c.o
[ 50%] Building C object src/CMakeFiles/r_433.dir/devices/intertechno.c.o
[ 51%] Building C object src/CMakeFiles/r_433.dir/devices/kedsum.c.o
[ 51%] Building C object src/CMakeFiles/r_433.dir/devices/kerui.c.o
[ 51%] Building C object src/CMakeFiles/r_433.dir/devices/klimalogg.c.o
[ 52%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse.c.o
[ 52%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_breezepro.c.o
[ 53%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_r1.c.o
[ 53%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_th3.c.o
[ 54%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_tx141x.c.o
[ 54%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_tx35.c.o
[ 55%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_wr1.c.o
[ 55%] Building C object src/CMakeFiles/r_433.dir/devices/lacrosse_ws7000.c.o
[ 56%] Building C object src/CMakeFiles/r_433.dir/devices/lacrossews.c.o
[ 56%] Building C object src/CMakeFiles/r_433.dir/devices/lightwave_rf.c.o
[ 56%] Building C object src/CMakeFiles/r_433.dir/devices/m_bus.c.o
[ 57%] Building C object src/CMakeFiles/r_433.dir/devices/markisol.c.o
[ 57%] Building C object src/CMakeFiles/r_433.dir/devices/marlec_solar.c.o
[ 58%] Building C object src/CMakeFiles/r_433.dir/devices/maverick_et73.c.o
[ 58%] Building C object src/CMakeFiles/r_433.dir/devices/maverick_et73x.c.o
[ 59%] Building C object src/CMakeFiles/r_433.dir/devices/mebus.c.o
[ 59%] Building C object src/CMakeFiles/r_433.dir/devices/missil_ml0757.c.o
[ 60%] Building C object src/CMakeFiles/r_433.dir/devices/new_template.c.o
[ 60%] Building C object src/CMakeFiles/r_433.dir/devices/newkaku.c.o
[ 61%] Building C object src/CMakeFiles/r_433.dir/devices/nexa.c.o
[ 61%] Building C object src/CMakeFiles/r_433.dir/devices/nexus.c.o
[ 61%] Building C object src/CMakeFiles/r_433.dir/devices/nice_flor_s.c.o
[ 62%] Building C object src/CMakeFiles/r_433.dir/devices/norgo.c.o
[ 62%] Building C object src/CMakeFiles/r_433.dir/devices/oil_standard.c.o
[ 63%] Building C object src/CMakeFiles/r_433.dir/devices/oil_watchman.c.o
[ 63%] Building C object src/CMakeFiles/r_433.dir/devices/opus_xt300.c.o
[ 64%] Building C object src/CMakeFiles/r_433.dir/devices/oregon_scientific.c.o
[ 64%] Building C object src/CMakeFiles/r_433.dir/devices/oregon_scientific_sl109h.c.o
[ 65%] Building C object src/CMakeFiles/r_433.dir/devices/oregon_scientific_v1.c.o
[ 65%] Building C object src/CMakeFiles/r_433.dir/devices/philips_aj3650.c.o
[ 66%] Building C object src/CMakeFiles/r_433.dir/devices/philips_aj7010.c.o
[ 66%] Building C object src/CMakeFiles/r_433.dir/devices/prologue.c.o
[ 66%] Building C object src/CMakeFiles/r_433.dir/devices/proove.c.o
[ 67%] Building C object src/CMakeFiles/r_433.dir/devices/quhwa.c.o
[ 67%] Building C object src/CMakeFiles/r_433.dir/devices/radiohead_ask.c.o
[ 68%] Building C object src/CMakeFiles/r_433.dir/devices/rftech.c.o
[ 68%] Building C object src/CMakeFiles/r_433.dir/devices/rojaflex.c.o
[ 69%] Building C object src/CMakeFiles/r_433.dir/devices/rubicson.c.o
[ 69%] Building C object src/CMakeFiles/r_433.dir/devices/rubicson_48659.c.o
[ 70%] Building C object src/CMakeFiles/r_433.dir/devices/s3318p.c.o
[ 70%] Building C object src/CMakeFiles/r_433.dir/devices/schraeder.c.o
[ 71%] Building C object src/CMakeFiles/r_433.dir/devices/scmplus.c.o
[ 71%] Building C object src/CMakeFiles/r_433.dir/devices/secplus_v1.c.o
[ 71%] Building C object src/CMakeFiles/r_433.dir/devices/secplus_v2.c.o
[ 72%] Building C object src/CMakeFiles/r_433.dir/devices/sharp_spc775.c.o
[ 72%] Building C object src/CMakeFiles/r_433.dir/devices/silvercrest.c.o
[ 73%] Building C object src/CMakeFiles/r_433.dir/devices/simplisafe.c.o
[ 73%] Building C object src/CMakeFiles/r_433.dir/devices/smoke_gs558.c.o
[ 74%] Building C object src/CMakeFiles/r_433.dir/devices/solight_te44.c.o
[ 74%] Building C object src/CMakeFiles/r_433.dir/devices/somfy_iohc.c.o
[ 75%] Building C object src/CMakeFiles/r_433.dir/devices/somfy_rts.c.o
[ 75%] Building C object src/CMakeFiles/r_433.dir/devices/springfield.c.o
[ 76%] Building C object src/CMakeFiles/r_433.dir/devices/steelmate.c.o
[ 76%] Building C object src/CMakeFiles/r_433.dir/devices/tfa_30_3196.c.o
[ 76%] Building C object src/CMakeFiles/r_433.dir/devices/tfa_30_3221.c.o
[ 77%] Building C object src/CMakeFiles/r_433.dir/devices/tfa_drop_30.3233.c.o
[ 77%] Building C object src/CMakeFiles/r_433.dir/devices/tfa_marbella.c.o
[ 78%] Building C object src/CMakeFiles/r_433.dir/devices/tfa_pool_thermometer.c.o
[ 78%] Building C object src/CMakeFiles/r_433.dir/devices/tfa_twin_plus_30.3049.c.o
[ 79%] Building C object src/CMakeFiles/r_433.dir/devices/thermopro_tp11.c.o
[ 79%] Building C object src/CMakeFiles/r_433.dir/devices/thermopro_tp12.c.o
[ 80%] Building C object src/CMakeFiles/r_433.dir/devices/thermopro_tx2.c.o
[ 80%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_abarth124.c.o
[ 81%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_citroen.c.o
[ 81%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_elantra2012.c.o
[ 81%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_ford.c.o
[ 82%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_hyundai_vdo.c.o
[ 82%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_jansite.c.o
[ 83%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_jansite_solar.c.o
[ 83%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_pmv107j.c.o
[ 84%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_renault.c.o
[ 84%] Building C object src/CMakeFiles/r_433.dir/devices/tpms_toyota.c.o
[ 85%] Building C object src/CMakeFiles/r_433.dir/devices/ts_ft002.c.o
[ 85%] Building C object src/CMakeFiles/r_433.dir/devices/ttx201.c.o
[ 86%] Building C object src/CMakeFiles/r_433.dir/devices/vaillant_vrt340f.c.o
[ 86%] Building C object src/CMakeFiles/r_433.dir/devices/visonic_powercode.c.o
[ 86%] Building C object src/CMakeFiles/r_433.dir/devices/waveman.c.o
[ 87%] Building C object src/CMakeFiles/r_433.dir/devices/wg_pb12v1.c.o
[ 87%] Building C object src/CMakeFiles/r_433.dir/devices/ws2032.c.o
[ 88%] Building C object src/CMakeFiles/r_433.dir/devices/wssensor.c.o
[ 88%] Building C object src/CMakeFiles/r_433.dir/devices/wt0124.c.o
[ 89%] Building C object src/CMakeFiles/r_433.dir/devices/wt450.c.o
[ 89%] Building C object src/CMakeFiles/r_433.dir/devices/x10_rf.c.o
[ 90%] Building C object src/CMakeFiles/r_433.dir/devices/x10_sec.c.o
[ 90%] Linking C static library libr_433.a
[ 90%] Built target r_433
Scanning dependencies of target data
[ 90%] Building C object src/CMakeFiles/data.dir/data.c.o
[ 91%] Building C object src/CMakeFiles/data.dir/abuf.c.o
[ 91%] Building C object src/CMakeFiles/data.dir/term_ctl.c.o
[ 92%] Linking C static library libdata.a
[ 92%] Built target data
Scanning dependencies of target rtl_433
[ 93%] Building C object src/CMakeFiles/rtl_433.dir/rtl_433.c.o
[ 93%] Linking C executable rtl_433
libr_433.a(sdr.c.o): In function `sdr_apply_settings':
sdr.c:(.text+0x1410): undefined reference to `rtlsdr_set_bias_tee'
collect2: error: ld returned 1 exit status
src/CMakeFiles/rtl_433.dir/build.make:97: recipe for target 'src/rtl_433' failed
make[2]: *** [src/rtl_433] Error 1
CMakeFiles/Makefile2:1109: recipe for target 'src/CMakeFiles/rtl_433.dir/all' failed
make[1]: *** [src/CMakeFiles/rtl_433.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by zuckschwerdt** [comment]
The 0.5.3 rtlsdr release is [nearly 8 years old](https://github.com/osmocom/rtl-sdr/commit/df9596b). And `rtlsdr_set_bias_tee()` [is a recent addition](https://github.com/osmocom/rtl-sdr/commit/2be1612), well 5 years ago ;)

**Comment by zuckschwerdt** [comment]
You can just delete that line (or the whole 4 lines `if`) with no adverse effect, compile should run fine then :)

**Comment by gvanem** [comment]
In Welle-IO, it's prototypes as:
```c
int __attribute__((weak)) rtlsdr_set_bias_tee(rtlsdr_dev_t *dev, int on);
```

**Comment by zuckschwerdt** [comment]
True, older or constrainted hardware is very likely Linux. I'm thinking something like this:
```
#include <rtl-sdr.h>
+ #if defined(__linux__) && (defined(__GNUC__) || defined(__clang__))
+ // not available in 0.5.3, allow weak link for Linux
+ int __attribute__((weak)) rtlsdr_set_bias_tee_xx(rtlsdr_dev_t *dev, int on);
+ #endif
```
To avoid a `warning: address of function 'rtlsdr_set_bias_tee' will always evaluate to 'true' [-Wpointer-bool-conversion]`, we need to `#ifdef` the `if` on the actual call too ;)
